sdk/keyvault/keyvault-secrets/tests/secrets.test.ts

```diff
@@ -0,0 +1,160 @@
+import * as assert from "assert";
+import { expect } from "chai";
+import {
+  getKeyvaultName,
+  getCredentialWithServicePrincipalSecret,
+  getUniqueName
+} from "./utils/utils.common";
+import { SecretsClient } from "../src";
+import * as msRestNodeAuth from "@azure/ms-rest-nodeauth";
+
+describe("Secret client", () => {
+  const clientId = process.env["AAD_CLIENT_ID"] || "";
+  const clientSecret = process.env["AAD_CLIENT_SECRET"] || "";
+  const tenantId = process.env["AAD_TENANT_ID"] || "";
+  const vaultName = process.env["KEYVAULT_NAME"] || "<keyvault-name>";
+
+  let client: SecretsClient;
+  let version: string;
+
+  before(async () => {
+    const url = `https://${vaultName}.vault.azure.net`;
+
+    const credential = await msRestNodeAuth.loginWithServicePrincipalSecret(
+      clientId,
+      clientSecret,
+      tenantId,
+      {
+        tokenAudience: "https://vault.azure.net"
+      }
+    );
+
+    client = new SecretsClient(url, credential);
+    version = "";
+  });
+
+  it("can add a secret", async () => {
+    const secretName = getUniqueName("secret");
+    const secretValue = getUniqueName("value");
+    after(async () => {
+      await client.deleteSecret(secretName);
+    });
+    const result = await client.setSecret(secretName, secretValue);
+
+    assert.equal(result.name, secretName, "Unexpected secret name in result from setSecret().");
+    assert.equal(result.value, secretValue, "Unexpected secret value in result from setSecret().");
+  });
+
+  it("can retrieve the latest version of a secret value", async () => {
+    const secretName = getUniqueName("secret");
+    const secretValue = getUniqueName("value");
+    after(async () => {
+      await client.deleteSecret(secretName);
+    });
+    await client.setSecret(secretName, secretValue);
+
+    const result = await client.getSecret(secretName);
+
+    assert.equal(result.name, secretName, "Unexpected secret name in result from setSecret().");
+    assert.equal(result.value, secretValue, "Unexpected secret value in result from setSecret().");
+  });
+
+  it("can set a secret with attributes", async () => {
+    const secretName = getUniqueName("secret");
+    const secretValue = getUniqueName("value");
+    const expiryDate = new Date("3000-01-01");
+    after(async () => {
+      await client.deleteSecret(secretName);
+    });
+    await client.setSecret(secretName, secretValue, { expires: expiryDate });
+
+    const updated = await client.getSecret(secretName);
+
+    // TODO: Investigate. The service seems to change the milliseconds part of the expiry date.
+    // For now just compare year, month, and date in assertion.
+    assert.ok(
+      updated.expires!.getUTCFullYear() === expiryDate.getUTCFullYear() &&
+        updated.expires!.getUTCMonth() === expiryDate.getUTCMonth() &&
+        updated.expires!.getUTCDate() === expiryDate.getUTCDate(),
+      "Expect attribute 'expires' to be updated."
+    );
```

Comment:
Are you positive the service is changing this milliseconds?  This field is actually serialized as a unix timestamp, so it might be that the serialization code has a bug in it.

Reply:
Mr Scott, I didn't write these tests, but I can make an issue to address your point. This PR is just about making sure the previously written tests pass CI (only secrets had tests before, the keys PR has new tests).